torchtune/data/_messages.py

```diff
@@ -254,12 +255,21 @@ def __call__(self, sample: Mapping[str, Any]) -> Mapping[str, Any]:
                 eot=True,
             ),
         ]
-        if self.new_system_prompt is not None:
+
+        system_prompt = None
+        if self.has_system and self.column_map["system"] in sample and sample[self.column_map["system"]]:
```

Comment:
I think we should do something to the effect of:
- If "system" is in column map, user has specified a system column that is not the default "system" -> use that to key into sample dict
- If "system" is in the sample dict, then the user is using the default system column name -> use that to key into sample dict directly. Column map can be none 
- If neither of these are true, there is no system column

This will keep the behavior I described above. Let me know if this makes sense or it is unnecessarily complicating things.